### PR TITLE
feat: if a policy matches the name of a model we automatically register it

### PR DESCRIPTION
**Pull request by valentindrdt**
| Q             | A
| ------------- | ---
| Branch?       | main
| Tickets       | Closes #..., closes #... 
| License       | MIT
| Doc PR        | api-platform/docs#... 

Automatic policy registration

### CODE REVIEW

File: src/Laravel/Security/ResourceAccessChecker.php [hunk 1]
```diff
@@ -13,13 +13,27 @@
...
+        if (($object = $extraVariables['object']) instanceof Paginator) {
+            // do not deny access if no items are found
+            if (0 === $object->count()) {
+                return true;
+            }
+
+            // we're only checking access to the first item
+            foreach ($object as $obj) {
+                return Gate::allows($expression, $obj);
+            }
```
Comment (dunglas) [inline src/Laravel/Security/ResourceAccessChecker.php:34]:
```suggestion
            // we're only checking access to the first item
            foreach ($object as $obj) {
                return Gate::allows($expression, $obj);
            }
            
            return true;
```

micro-optim (to be tested)

**Review by dunglas — APPROVED** [review]
Very cool!

**Comment by dunglas** [comment]
Couldn't we do something similar for Form Request btw?

File: src/Laravel/Security/ResourceAccessChecker.php (hunk 1 above)
Comment (dunglas) [inline src/Laravel/Security/ResourceAccessChecker.php:34]:
```suggestion
        // ugly way to handle collection
        if (($object = $extraVariables['object']) instanceof Paginator) {
            Gate::allows($expression);
```

This should be enough.

Comment (valentindrdt) [reply src/Laravel/Security/ResourceAccessChecker.php:34]: I refactored this part, therefore this code is outdated

**Pull request merged**